core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -314,34 +314,25 @@ object TestUtils extends Logging {
                        codec: CompressionType = CompressionType.NONE,
                        timestamp: Long = RecordBatch.NO_TIMESTAMP,
                        magicValue: Byte = RecordBatch.CURRENT_MAGIC_VALUE): MemoryRecords = {
-    records(magicValue = magicValue, codec = codec, (key, value, timestamp))
+    records(Seq(new SimpleRecord(timestamp, key, value)), magicValue = magicValue, codec = codec)
   }
 
   def recordsWithValues(magicValue: Byte,
                         codec: CompressionType,
                         values: Array[Byte]*): MemoryRecords = {
-    records(magicValue, codec, values.map((null, _, RecordBatch.NO_TIMESTAMP)): _*)
+    records(values.map(value => new SimpleRecord(value)), magicValue, codec)
   }
 
-  def records(magicValue: Byte,
-              codec: CompressionType,
-              records: (Array[Byte], Array[Byte], Long)*): MemoryRecords = {
-    this.records(records, magicValue = magicValue, codec = codec)
```

Comment:
I was going to suggest removing this, but was unsure. Happy that you did it anyway. :)